src/applications/soundDetectionDemo/robotMovement/src/vRobotMovement.cpp

```diff
@@ -0,0 +1,369 @@
+
+private:
+    // Input port: type AE
+    vReadPort< vector< AE > > input_port;
+    // Output port: type Bottle
+    BufferedPort<Bottle> output_port;
+
+    // ----------------------------------------------------------------------------
+    //                              JOINT CONTROL
+    // Joint driver
+    PolyDriver joint_driver;
+
+    // Limits of the joint
+    IControlLimits *joint_limits;
+
+    // Enconder of the joint
+    IEncoders *joint_enconder;
+
+    // Control mode of the joint
+    IControlMode *joint_control_mode;
+
+    // Position control of the joint
+    IPositionControl *joint_position_control;
+    // ----------------------------------------------------------------------------
+
+    // Debug flag
+    // If activated, the module shows info about the classification in the terminal
+    bool is_debug_flag;
+
+    // Simulation flag
+    // If activated, it means we are using the simulation platform
+    bool is_simulation;
+
+    // Number of sound source positions. This number must be the same as the number 
+    // of output neurons used in vAuditoryAttention sound sources population
+    int number_sound_source_neurons;
+
+    // Body part
+    std::string body_part;
+
+    // Joint ID
+    int joint_id;
+
+    // Joint speed
+    double joint_speed;
+
+    // Joint max limit
+    double joint_max_limit;
+    double joint_max_limit_reduced;
+
+    // Joint min limit
+    double joint_min_limit;
+    double joint_min_limit_reduced;
+
+    // Joint range
+    double joint_range;
+
+    // Joint steep
+    double joint_step;
+    double half_joint_step;
+
+    // Joint initial position
+    double joint_initial_position;
+
+    // Joint current position
+    double joint_current_position;
+
+    // Joint desired position
+    double joint_desired_position;
+
+    // Period
+    double update_period;
+
+public:
+
+    vRobotMovement() {}
+
+    virtual bool configure(yarp::os::ResourceFinder& rf)
+    {
+        yInfo() << "Configuring the module...";
+
+        // Set the module name used to name ports
+        setName((rf.check("name", Value("/vRobotMovement")).asString()).c_str());
+
+        // Open input port
+        yInfo() << "Opening input port...";
+        if(!input_port.open(getName() + "/AE:i")) {
+            yError() << "Could not open input port";
+            return false;
+        }
+        // Open output port
+        yInfo() << "Opening output port...";
+        if(!output_port.open(getName() + "/pos:o")) {
+            yError() << "Could not open output port";
+            return false;
+        }
+
+        // Read flags and parameters
+        
+        // Debug flag: flase by default
+        is_debug_flag = rf.check("is_debug_flag") &&
+                rf.check("is_debug_flag", Value(true)).asBool();
+        yInfo() << "Flag is_debug_flat is: " << is_debug_flag;
+
+        // Simulation flag: true by default
+        bool default_is_simulationr = true;
```

Comment:
this is incorrect use of the logic functions. If you want a boolean that is default true, just use `bool is_simulatoin = rf.check("is_simulation", Value(True))).asBool();`. What you are doing with the logic functions isn't what you want. This comment goes for any other place you are trying to do the same. fyi - the way you are doing it is very misleading in the actual outcome of the flag.